develop/2017-06-20-rodolfoviana-expenses-on-car-rental.py

```diff
@@ -0,0 +1,143 @@
+# 
+# ~~**First step:** get a list of congresspersons, the amount reimbursed by them since Jan. 2015 and the dates of those reimbursements. Then we cross these data with the list of companies that rented those vehicles so we can get information on where those rentals occurred.~~ *Done!*
+# 
+# **Second step:** get datasets (sessions, speeches) that may prove whether congressperson was or was not in DF in specific periods of time: when the vehicles were rented. So we can get, as a result, months in which the congressperson spent most of his/her time in DF, but payed a full-month rent somewhere else. *I need some help here.*
+# 
+# -- Rodolfo Viana
+
+# In[1]:
+
+import pandas as pd
+import numpy as np
+
+data = pd.read_csv('../data/2017-06-04-reimbursements.xz',
+                      dtype={'applicant_id': np.str,
+                             'cnpj_cpf': np.str,
+                             'congressperson_id': np.str,
+                             'congressperson_name': np.str,
+                             'subquota_number': np.str,
+                             'issue_date': np.str,
+                             'document_id': np.str},
+                      low_memory=False)
+
+
+# #### Selecting term, subquota
+# 
+# There are 14,263 reimbursements since Jan. 2015. They sum up to R$ 60,373,960.80.
+
+# In[2]:
+
+data = data[data['year'] >= 2015]
+data = data[data['subquota_description'] == 'Automotive vehicle renting or charter']
+data['cnpj_cpf'] = data['cnpj_cpf'].str.replace(r'[\.\/\-]', '')
+data.subquota_description.value_counts()
+
+
+# In[3]:
+
+data.net_values.sum()
+
+
+# In[4]:
+
+congressperson_list = data[['congressperson_name', 
+                            'congressperson_id', 
+                            'net_values', 
+                            'month', 
+                            'year', 
+                            'issue_date', 
+                            'document_id',
+                            'cnpj_cpf']]
+
+
+# In[5]:
+
+congressperson_expenses = congressperson_list.groupby(['congressperson_name', 
+                                                       'year', 
+                                                       'month', 
+                                                       'issue_date', 
+                                                       'document_id']).agg({'net_values':sum})
+
+congressperson_expenses.head(20)
+
+
+# #### Getting companies dataset, excluding those from DF, merging with reimbursements dataset
+# 
+# There are 10,945 companies outside DF. The receipts sum up to R$ 49,156,001.56.
+
+# In[6]:
+
+companies = pd.read_csv('../data/2017-05-21-companies-no-geolocation.xz', low_memory=False)
+companies = companies[companies['state'] != 'DF']
+companies['cnpj'] = companies['cnpj'].str.replace(r'[\.\/\-]', '')
+
+
+# In[7]:
+
+dataset = pd.merge(data, companies, how='inner',
```

Comment:
Not sure if you're aware of it, but when you do an inner join/merge, rows in `data` without a CNPJ in `companies` and the opposite, companies without a corresponding `CNPJ_CPF` in `data`, will be removed from the result (`dataset`). Since `2017-05-21-companies-no-geolocation.xz` accords to about 99% of the companies in the reimbursements until that date, you're probably not removing many records, if any.

[A picture is worth a thousand words](https://stackoverflow.com/a/20298671).